### ISSUE TITLE
Support user-defined name types

### ISSUE THREAD

**Issue by WGroleau** [fisharebest/webtrees#4735]
Spec 5.5.1 for NAME:TYPE has a list of options ending with
user_defined = other text name that defines the name type.
webtrees should provide a way for users to enter user_defined values in a free-set field and/or the PHRASE of GEDCOM v7

Similar may apply to 
NAME:ROMN
PLAC:ROMN
PLAC:FONE
NAME:FONE
MARR:TYPE
INDI:ASSO:TYPE 
INDI:EVEN:_ASSO:TYPE

See https://www.webtrees.net/index.php/en/forum/9-request-for-new-feature/37718-user-defined-name-type#97578

**Comment by zhoueysun** [comment]
I have been working on this for over 5 years (https://www.webtrees.net/index.php/en/forum/help-for-2-0/34969-customising-marriage-name-type-dropdown-list-v2-x) and my attempts to write a custom module has failed. The simplest for the non-programmer (me) was always to just modify the php file what Greg wrote.
Having the option of the user inputting free text would be most useful. However, generally the webtrees administrators would like a list of certain TYPE choices defined for the user. It would be an added benefit if the administrator can modify the default list of TYPES

**Comment by WGroleau** [comment]
I would agree that "It would be an added benefit if the administrator can modify the default list of TYPES" but what a mess we'd have if we ever expected administrators to be able to edit every menu in the whole system.

**Comment by AndreasRauhut** [comment]
Modifying Gregs list in the php would not be a good choice for the admin, because he needs also the terms to be translated.

I would be glad to only add **midwife** to the ASSO/_ASSO types

**Comment by allan-wtf** [comment]
I had been using user defined types in the previous release of WebTrees (v1.x & v.2.0.x) and had implemented the fix suggested by Greg several years earlier (28226-name-type); however with release of 2.1 I lost the translations of my user-defined name types.  The import of user-defined name types (gedcom import) still functions in original english, but these are no longer accessible for online-entry / name type entry within webtrees menu, however can be maintained with the RAW Gedcom.  However, the loss of translations is the real issue.

The fix required is in two parts : extend webtrees (possible CustomNameType module) to :
1) recognize selected user-defined name types to be added to the NAME.TYPE pull down menu (nice to have)
2) allow user provided translations of the selected name types for inclusion within module I18N (need to address)

Further to Zhoueysun's comment, for my wife's family we have been recording the / childhood / adult / honorific / names present in Chinese family records, and in order to address the modern day migration I have included "western" as a name distinct from the ROMN (pinyin) name type.

My approach to the use of "generation" as a name element is also broken, but that will be a different fix which I am revisiting.
